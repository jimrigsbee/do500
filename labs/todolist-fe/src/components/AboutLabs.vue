<template>
  <div>
    <h3>Learn more about Open Innovation Labs</h3>
    <ul>
      <li v-for="item in aboutLabs" :key="item.name" ><a :href="item.url" target="_blank">{{item.name}}</a></li>
    </ul>
    <h3>Connect with us</h3>
    <ul>
      <li v-for="item in social" :key="item.name" ><a :href="item.url" target="_blank">{{item.name}}</a></li>
    </ul>
  </div>
</template>

<script>
// @ is an alias to /src
export default {
  name: "AboutLabs",
  components: {},
  data() {
    return {
      aboutLabs: [
        {
          name: "Labs Homepage",
          url:
            "https://www.redhat.com/en/engage/open-innovation-labs-20170905?sc_cid=701f2000000tvPOAAY"
        },
        {
          name: "Lab Locations",
          url:
            "https://www.redhat.com/en/services/consulting/open-innovation-labs/locations"
        },
        {
          name: "Labs Tech",
          url: "https://www.redhat.com/en/explore/my-open-innovation-lab-stack"
        },
        {
          name: "Inside a Residency Blog",
          url:
            "https://developers.redhat.com/blog/2018/01/26/part-1-inside-open-innovation-labs-residency/"
        }
      ],
      social: [
        {
          name: "YouTube",
          url: "https://www.youtube.com/watch?v=RxOKoqpDuIA"
        },
        {
          name: "Twitter",
          url: "https://twitter.com/redhatlabs?lang=en"
        }
      ]
    };
  }
};
</script>


<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #a30000;
}
</style>
