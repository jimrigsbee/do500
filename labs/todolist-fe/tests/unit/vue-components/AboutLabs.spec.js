/* eslint-disable */
import { shallow } from "@vue/test-utils";
import AboutLabs from "@/components/AboutLabs.vue";
import * as all from "../setup.js";

describe("AboutLabs.vue", () => {
  it("has the expected html structure", () => {
    const wrapper = shallow(AboutLabs);
    expect(wrapper.element).toMatchSnapshot();
  });
});
