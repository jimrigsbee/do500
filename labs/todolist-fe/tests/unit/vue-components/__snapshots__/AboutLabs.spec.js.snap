// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutLabs.vue has the expected html structure 1`] = `
<div>
  <h3>
    Learn more about Open Innovation Labs
  </h3>
   
  <ul>
    <li>
      <a
        href="https://www.redhat.com/en/engage/open-innovation-labs-20170905?sc_cid=701f2000000tvPOAAY"
        target="_blank"
      >
        Labs Homepage
      </a>
    </li>
    <li>
      <a
        href="https://www.redhat.com/en/services/consulting/open-innovation-labs/locations"
        target="_blank"
      >
        Lab Locations
      </a>
    </li>
    <li>
      <a
        href="https://www.redhat.com/en/explore/my-open-innovation-lab-stack"
        target="_blank"
      >
        Labs Tech
      </a>
    </li>
    <li>
      <a
        href="https://developers.redhat.com/blog/2018/01/26/part-1-inside-open-innovation-labs-residency/"
        target="_blank"
      >
        Inside a Residency Blog
      </a>
    </li>
  </ul>
   
  <h3>
    Connect with us
  </h3>
   
  <ul>
    <li>
      <a
        href="https://www.youtube.com/watch?v=RxOKoqpDuIA"
        target="_blank"
      >
        YouTube
      </a>
    </li>
    <li>
      <a
        href="https://twitter.com/redhatlabs?lang=en"
        target="_blank"
      >
        Twitter
      </a>
    </li>
  </ul>
</div>
`;
